pkg/util/strategicpatch/patch.go

```diff
@@ -100,6 +100,11 @@ func mergeMap(original, patch map[string]interface{}, t reflect.Type) (map[strin
 			original[k] = patchV
 			continue
 		}
+		// Find whether the data type is a pointer or not and changing the type accordingly.
```

Comment:
Minor phrasing nit, change to: // If the data type is a pointer, resolve the element.